File: R/reactiveJsonEdit.R
reactiveJsonEditOutput <- function(outputId = NULL, label = NULL, height = "400px",
                                   width = "100%") {
  
  script <- 'document.getElementById("button_id").onclick = function() {
      var listdata = HTMLWidgets.find("#outputId").editor.getText();
       Shiny.onInputChange("edit_id", listdata);
      };'
  script <- script %>% 
    str_replace("button_id", paste0(outputId, "_save")) %>% 
    str_replace("outputId", outputId) %>% 
    str_replace("edit_id", paste0(outputId, "_edit"))
  
  tagList(
    jsoneditOutput(
      outputId = outputId,
      height = height,
      width = width
    ),
    tags$br(),
    actionButton(
      inputId = paste0(outputId, "_save"),
      label = label
    ),
    htmltools::tags$script(script)
  )
  
}

File: www/reactiveJsonEdit.js
function getRandomInt(max) {
  return Math.floor(Math.random() * max);
}

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
};

async function reactiveJsonEditMod(e) {
  await sleep(1000);
  var buttonWidget = '<button id="' + e.name + "_button" + '" type="button" style="background-image: none; width: 55px;">Guardar</button>';
  var nextSelector = "#" + e.name + " div.jsoneditor div.jsoneditor-menu div.jsoneditor-modes";
  jQuery(buttonWidget).insertAfter(nextSelector);
  document.getElementById(e.name + "_button").onclick = function() {
    var listdata = HTMLWidgets.find("#" + e.name).editor.getText();
    Shiny.onInputChange(e.name + "_edit", listdata);
    Shiny.onInputChange(e.name + "_save", getRandomInt(100));
  };
};

